src/components/_helpers/transferProps.js

```diff
@@ -1,7 +1,46 @@
-export const transferProps = ({
-  children,
-  className,
-  ref,
-  staticContext,
-  ...restProps
-}) => restProps;
+/**
+ * Controls passing of props from the React component to the HTML element
+ *
+ * Sometimes it is useful to have a mechanism to pass props from the React component to a rendered HTML element.
+ * It enables making the component interactive and helps improve its accessibility. However some props should
+ * never be passed to the HTML element as it would break things. This function is used to filter out such props.
+ *
+ * When run in development mode, the function will log the error to the console if any invalid props are passed.
+ *
+ * @param props The props that were passed to the React component
+ * @returns The props to be passed to the HTML element
+ */
+export const transferProps = (props) => {
+  const {
+    children,
+    className,
+    contentEditable,
+    dangerouslySetInnerHTML,
+    ref,
+    staticContext,
+    style,
+    suppressContentEditableWarning,
+    ...restProps
+  } = props;
+
+  if (process.env.NODE_ENV !== 'production') {
+    const invalidProps = [
+      'children',
+      'className',
+      'contentEditable',
+      'dangerouslySetInnerHTML',
+      'ref',
+      'staticContext',
+      'style',
+      'suppressContentEditableWarning',
```

Comment:
OK, here we go:

```suggestion
      'children', // We only read children from between the component tags, e.g. `<Component>children</Component>`.
      'className', // We want to control the styling ourselves and not allow anyone else to change it.
      'contentEditable', // ???
      'dangerouslySetInnerHTML', // We don't want the HTML of our components to be changed from outside.
      'ref', // ???
      'staticContext', // ???
      'style', // We don't want to allow manual styling of our components. Also, we use it to set custom properties in some places.
      'suppressContentEditableWarning', // ???
```

I'm just unable to put down the reasons for all of the items. This is all I'm aware of.

Reply:
done